executor/infoschema_reader.go

```diff
@@ -1084,3 +1086,35 @@ func (e *memtableRetriever) setDataFromSessionVar(ctx sessionctx.Context) error
 	e.rows = rows
 	return nil
 }
+
+// dataForAnalyzeStatusHelper is a helper function which can be used in show_stats.go
+func dataForAnalyzeStatusHelper(sctx sessionctx.Context) (rows [][]types.Datum) {
```

Comment:
this function is only called by `setDataForAnalyzeStatus()`, it's not necessary to extract it out of `setDataForAnalyzeStatus()`